src/guides/anatomy-of-an-http-transaction.md

```diff
@@ -0,0 +1,410 @@
+`response`. We'll get to those shortly.
+
+In order to have this thing actually serving requests, the [`listen`][] method
+needs to be called on the `server` object. In most cases, all you'll need for
+arguments is the port number you want the server to listen on. There are some
+other options too, so consult the [API reference][].
+
+## Method, URL and Headers
+
+When handling a request, the first thing you'll probably want to do is look at
+the method and URL, so that appropriate actions can be taken. Node makes this
+relatively painless by putting handy properties onto the `request` object.
+
+```javascript
+var method = request.method;
+var url = request.url;
+```
+> **Note:** The `request` object is an instance of [`IncomingMessage`][].
+
+The `method` here will always be a normal HTTP method/verb. The `url` is the
+full URL without the server, protocol or port. For a typical URL, this means
+everything after and including the third forward slash.
+
+Headers are also not far away. They're in their own object on `request` called
+`headers`.
+
+```javascript
+var headers = request.headers;
+var userAgent = headers['user-agent'];
+```
+
+It's important to note here that all headers are represented in lower-case only,
+regardless of how the client actually sent them. This helps to simplify the task
+of parsing headers for whatever purpose.
+
+If some headers are repeated, then their values are overwritten or joined
+together as comma-separated stings, depending on the header. In some cases, this
+can be problematic, so [`rawHeaders`][] is also available.
+
+## Request Body
+
+Many HTTP requests, like `POST` or `PUT` for example, might have a body as part
+of the request.
+
+To get at the body data, it's a little different from accessing headers. The
+`request` object that's passed in to a handler implements the
+[`ReadableStream`][] interface. This stream can be listened to or piped elswhere
+just like any other stream. We can grab the data right out of the stream by
+listening to the stream's `'data'` and `'end'` events.
+
+The chunk emitted in each `'data'` event is a [`Buffer`][], but if you know that
+you will only be dealing with UTF-8 strings and no binary data, it makes sense
+to let JavaScript coerce the chunks into strings.
+
+```javascript
+var body = '';
+request.on('data', function(chunk) {
+  body += chunk;
+}).on('end', function() {
+  // at this point, `body` has the entire request body stored in it
+});
+```
+
+> **Note:** This may seem a tad tedious, and in many cases, it is. Luckily,
+there are modules like [`body`][] on [`npm`][] which can help hide away some of
+this logic. It's important to have a good understanding of what's going on
+before going down that road, and that's why you're here!
+
+## A Quick Thing About Errors
+
+Since the `request` object is a [`ReadableStream`][], it's also an
+[`EventEmitter`][]. As such, it behaves in a slightly different way when an
```

Comment:
I really like that the necessity of handling errors is called out here.

I might rephrase "behaves in a slightly different way" because there's no base case for it to be different from — so it kind of causes a brief hiccup when reading ("Different from what? Did I miss something?")